kubespawner/reflector.py

```diff
@@ -48,7 +81,7 @@ def __init__(self, *args, **kwargs):
             config.load_incluster_config()
         except config.ConfigException:
             config.load_kube_config()
-        self.api = client.CoreV1Api()
+        self.api = getattr(client, self.api_group_name)()
 
         # FIXME: Protect against malicious labels?
```

Comment:
Do we also need to protect against malicious targets (earlier in this PR)?

Reply:
I don't think so right now, since these can only be specified by users editing jupyterhub_config.py files. I think we should take a pass in a separate PR doing more validation here, but for now I think this is fine.